src/altinn-app-frontend/src/features/expressions/validation.ts

```diff
@@ -84,18 +84,26 @@ function validateFunctionArgs(
   path: string[],
 ) {
   const expected = ExprFunctions[func].args;
+  const maxIdx = Math.max(expected.length, actual.length);
+  for (let idx = 0; idx < maxIdx; idx++) {
+    validateFunctionArg(func, idx, actual, ctx, path);
+  }
+}
+
+function validateFunctionArgLength(
+  func: ExprFunction,
+  actual: (BaseValue | undefined)[],
+  ctx: ValidationContext,
+  path: string[],
+) {
+  const expected = ExprFunctions[func].args;
```

Comment:
I'm just separating out a function for validating the argument length, so that functions supplying their own validator (specifically the `if` function) can skip the length-check, but keep the argument type validator.